functional/FunctionalTest.java

```diff
@@ -3813,6 +3818,8 @@ public static void main(String[] args) throws Exception {
     replicationRole = System.getenv("MINIO_JAVA_TEST_REPLICATION_ROLE");
     replicationBucketArn = System.getenv("MINIO_JAVA_TEST_REPLICATION_BUCKET_ARN");
 
+    isRunOnFail = System.getenv("RUN_ON_FAIL") != null && System.getenv("RUN_ON_FAIL").equals("1");
```

Comment:
This flag is only useful for `mint` mode. You could move this assignment line #3810